modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/resolver/RegistryLookupValueResolver.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.runtime.resolver;
+
+import static org.mule.util.Preconditions.checkArgument;
+import org.mule.api.MuleContext;
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleException;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Implementation of {@link ValueResolver} which accesses the mule registry
+ * and returns the value associated with {@link #key}.public static final
+ * <p/>
+ * Because the registry is mutable, {@link #isDynamic()} will always return
+ * {@code true} even though the odds are that the same value will always be returned.
+ * In case you want to prevent accessing the registry every time (this requires
+ * some confidence on your end about the registry value not changing) you could use
+ * this resolver in conjunction with {@link CachingValueResolverWrapper}
```

Comment:
ValueResolver is not part of the extension-api, it's part of the implementation. THe user does not say he wants to use this, the implementation uses it when right